### PR TITLE
chore: upgrade webpack-dev-server to v4.5.0

### PR DESCRIPTION
**Pull request by lex111**
## Motivation

See https://github.com/webpack/webpack-dev-server/releases/tag/v4.5.0

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

Try to run dev server locally.

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/facebook/docusaurus, and link to your PR here.)

### CODE REVIEW

File: packages/docusaurus/src/commands/start.ts [hunk 1]
```diff
@@ -8,7 +8,7 @@
 import {normalizeUrl, posixPath} from '@docusaurus/utils';
 import chalk = require('chalk');
 import chokidar from 'chokidar';
-
+import {ServerOptions} from 'https';
```
Comment (armano2) [inline packages/docusaurus/src/commands/start.ts:11]:
```suggestion
import type {ServerOptions} from 'https';
```

File: packages/docusaurus/src/commands/start.ts [hunk 2]
```diff
@@ -197,6 +197,12 @@ export default async function start(
         ...{pollingOptions},
       },
     })),
+    ...(httpsConfig && {
+      server: {
+        type: 'https',
+        options: httpsConfig as ServerOptions,
```
Comment (Josh-Cena) [inline packages/docusaurus/src/commands/start.ts:203]: Our `getHttpsConfig` returns `true` if the env says `HTTPS=true` but doesn't provide the key and cert. Is that actually a legitimate situation to be in? Should we actually return `false | ServerOptions` instead, and throw an error if there's no key?

Comment (lex111) [reply packages/docusaurus/src/commands/start.ts]: Yes it is valid, in this case a self-signed certificate will be used, just fixed it.

**Review by slorber — APPROVED** [review]
👍 we can merge this if you think it's ready

we'll figure out possible issues while working on Docusaurus until next release

**Pull request merged**